tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -329,20 +375,60 @@ private void AssertAuthenticationFails(MongoClientSettings settings)
         {
             return CreateDatabaseUser(client, "admin", userName, password, "readWriteAnyDatabase", mechanisms);
         }
-        
+
         private BsonDocument CreateDatabaseUser(
             MongoClient client,
             string source,
-            string userName, 
-            string password, 
-            string role, 
+            string userName,
+            string password,
+            string role,
             params string[] mechanisms)
         {
             var createUserCommand =
-                $"{{createUser: '{userName}', pwd: '{password}'," 
+                $"{{createUser: '{userName}', pwd: '{password}',"
                 + $"roles: ['{role}'], mechanisms: {mechanisms.ToJson()}}}";
             return client.GetDatabase(source).RunCommand<BsonDocument>(createUserCommand);
         }
-        
+
+        private BsonDocument CreateX509DatabaseUser(MongoClient client, string userName)
+        {
+            var createUserCommand =
+                $"{{ createUser : '{userName}'," +
+                @"   roles : [ { role : 'readWrite', db : 'test' },
+                               { role : 'userAdminAnyDatabase', db : 'admin' } ]
+                     writeConcern: { w : 'majority', j : true }
+                  }";
+            return client.GetDatabase("$external").RunCommand<BsonDocument>(createUserCommand);
+        }
+
+        private List<object> DropDatabaseUser(MongoClient client, string userName, IEnumerable<string> databases)
+        {
+            var dropUserCommand =
+                $"{{ dropUser : '{userName}'," +
+                @"   writeConcern: { w : 'majority', j : true }
+                  }";
+
+            return databases.Select(TryDropUser).ToList();
+
+            object TryDropUser(string database)
+            {
+                object result;
+                try
+                {
+                    result = client.GetDatabase(database).RunCommand<BsonDocument>(dropUserCommand);
+                }
+                catch (MongoCommandException ex) // command throws if user doesn't exist
+                {
+                    result = ex;
+                }
+                return result;
+            }
```

Comment:
I don't see why this method needs to take a list of databases. It is only ever called with just one database.

The whole helper method can be simplified to just:

    private object DropDatabaseUser(MongoClient client, string database, string userName)
    {
        var dropUserCommand =
            $"{{ dropUser : '{userName}'," +
            @"   writeConcern: { w : 'majority', j : true }
              }";

        try
        {
            return client.GetDatabase(database).RunCommand<BsonDocument>(dropUserCommand);
        }
        catch (MongoCommandException ex) // command throws if user doesn't exist
        {
            return ex;
        }
    }

You should also consider changing the return type to `void` since no one looks at the return value.

Reply:
I've taken your refactoring. As before, I feel having this method return a value could be useful in for future tests or for debugging so I'm inclined to leave it unless people feel very strongly about this.

Reply:
After some additional thought, I've made this method void :)